ucp/core.py

```diff
@@ -531,30 +573,35 @@ async def close(self):
                 return
             self._shutting_down_peer = True
 
-            # Send a shutdown message to the peer
-            msg = CtrlMsg.serialize(opcode=1, close_after_n_recv=self._send_count)
-            msg_arr = Array(msg)
-            log = "[Send shutdown] ep: %s, tag: %s, close_after_n_recv: %d" % (
-                hex(self.uid),
-                hex(self._ctrl_tag_send),
-                self._send_count,
-            )
-            logger.debug(log)
-            try:
-                await comm.tag_send(
-                    self._ep, msg_arr, msg_arr.nbytes, self._ctrl_tag_send, name=log,
-                )
-            # The peer might already be shutting down thus we can ignore any send errors
-            except UCXError as e:
-                logging.warning(
-                    "UCX failed closing worker %s (probably already closed): %s"
-                    % (hex(self.uid), repr(e))
+            if self._use_tags:
```

Comment:
```suggestion
            if self._tags:
```

Reply:
You will also have to change all the other `self._msg_tag_*` and `self._ctrl_tag_*` accesses  to dict lookup